k8s/agent-operator/src/main/java/com/walmartlabs/concord/agentoperator/scheduler/Scheduler.java

```diff
@@ -154,7 +154,10 @@ private void updateTargetSize(AgentPoolInstance i) throws IOException {
         }
 
         int queueQueryLimit = i.getResource().getSpec().getQueueQueryLimit();
-        List<ProcessQueueEntry> queueEntries = processQueueClient.query("ENQUEUED", queueQueryLimit);
+        Map<String, Object> queueSelector = i.getResource().getSpec().getQueueSelector();
+        String flavor = mapHasKey(queueSelector,"agent") ? mapHasKey(queueSelector.get("agent"),"flavor") ?
```

Comment:
is it possible that `queueSelector.agent` will be not map?
I think it will always a map. 
so you can use `ConfigurationUtils.get(queueSelector, "agent", "flavor")`